mlflow/utils/file_utils.py

```diff
@@ -588,6 +595,108 @@ def download_file_using_http_uri(http_uri, download_path, chunk_size=100000000,
+    then reconstructs the file from the downloaded chunks. This allows for downloads of large files
+    without OOM risk.
+
+    Note : This function is meant to download files using presigned urls from various cloud
+            providers.
+    Returns a dict of chunk index : exception, if one was thrown for that index.
+    """
+    num_requests = int(math.ceil(file_size / float(chunk_size)))
+    # Create file if it doesn't exist or erase the contents if it does. We should do this here
+    # before sending to the workers so they can each individually seek to their respective positions
+    # and write chunks without overwriting.
+    open(download_path, "w").close()
+    starting_index = 0
+    if uri_type == ArtifactCredentialType.GCP_SIGNED_URL or uri_type is None:
+        # GCP files could be transcoded, in which case the range header is ignored.
+        # Test if this is the case by downloading one chunk and seeing if it's larger than the
+        # requested size. If yes, let that be the file; if not, continue downloading more chunks.
+        download_chunk(
+            range_start=0,
+            range_end=chunk_size - 1,
+            headers=headers,
+            download_path=download_path,
+            http_uri=http_uri,
+        )
+        downloaded_size = os.path.getsize(download_path)
+        # If downloaded size was equal to the chunk size it would have been downloaded serially,
+        # so we don't need to consider this here
+        if downloaded_size > chunk_size:
+            return {}
+        else:
+            starting_index = 1
+
+    failed_downloads = {}
+
+    def run_download(range_start, range_end, index):
+        range_start = index * chunk_size
+        range_end = range_start + chunk_size - 1
+
+        with tempfile.TemporaryFile(mode="w+") as temp_file:
+            download_proc = _exec_cmd(
+                cmd=[
+                    sys.executable,
+                    download_cloud_file_chunk.__file__,
+                    "--range-start",
+                    range_start,
+                    "--range-end",
+                    range_end,
+                    "--headers",
+                    json.dumps(headers or {}),
+                    "--download-path",
+                    download_path,
+                    "--http-uri",
+                    http_uri,
+                    "--temp-file",
+                    temp_file,
+                ],
+                throw_on_error=True,
+                synchronous=False,
+                capture_output=True,
+                stream_output=False,
+                env=env,
+            )
+            return_code = download_proc.wait()
+            if return_code != 0:
+                temp_file.seek(0)
+                file_contents = temp_file.read()
+                if file_contents:
+                    return json.loads(file_contents)
+                else:
+                    _, stderr = download_proc.communicate()
```

Comment:
```suggestion
            _, stderr = download_proc.communicate()
            if rdownload_proc.eturn_code != 0:
                temp_file.seek(0)
                file_contents = temp_file.read()
                if file_contents:
                    return json.loads(file_contents)
```

We can replace `wait` with `communicate`.